Explorer/Assets/DCL/CharacterMotion/Platforms/PlatformRaycast.cs

```diff
@@ -0,0 +1,38 @@
+﻿using CrdtEcsBridge.Physics;
+using DCL.CharacterMotion.Components;
+using DCL.CharacterMotion.Settings;
+using System.Runtime.CompilerServices;
+using UnityEngine;
+
+namespace DCL.CharacterMotion.Platforms
+{
+    public static class PlatformRaycast
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void Execute(CharacterPlatformComponent platformComponent, float radius, Transform transform, ICharacterControllerSettings settings)
+        {
+            float rayDistance = settings.PlatformRaycastLength;
+            float halfDistance = (rayDistance * 0.5f) + radius;
+
+            var ray = new Ray
+            {
+                origin = transform.position + (Vector3.up * halfDistance),
+                direction = Vector3.down,
+            };
+
+            bool hasHit = Physics.SphereCast(ray, radius, out RaycastHit hitInfo, rayDistance + radius, PhysicsLayers.CHARACTER_ONLY_MASK);
```

Comment:
what about `Physics.SphereCastNonAlloc` or it doesn't make sense in this context?

Reply:
![image](https://github.com/decentraland/unity-explorer/assets/7646450/df1437cd-8689-4a5d-b0f7-fd19f455130b)
`SphereCast` already has 0 allocations, `Physics.SphereCastNonAlloc` is a variant of `SphereCastAll` which allocates